packages/x-components/build/rollup.config.ts

```diff
@@ -87,6 +87,11 @@ export const rollupConfig = createRollupOptions({
       // lang is set to ts:
       // https://github.com/vuejs/rollup-plugin-vue/issues/272#issuecomment-491721842
       needMap: false,
+      template: {
+        compilerOptions: {
+          whitespace: 'condense'
+        }
+      } as VuePluginOptions['template'],
```

Comment:
```typescript
vue({
    css: false,
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore Undocumented option to disable vue sourcemap generation because it breaks if
    // lang is set to ts:
    // https://github.com/vuejs/rollup-plugin-vue/issues/272#issuecomment-491721842
    needMap: false,
    template: {
      compilerOptions: {
        whitespace: 'condense'
      }
    },
    style: {
      postcssCleanOptions: { disabled: true }
    }
  } VuePluginOptions)
```

WDYT? 🤔

Reply:
That change turns the `template.compiler` property to be required. I don't know why with @javieri-empathy approach it doesn't require `template.compiler` to be defined, as he is using the very same type. Also, the type is already properly inferred, there's no need to type assert it. I believe @javieri-empathy did it to avoid the issue I was talking before.